### ISSUE TITLE
Expecting a parameter declaration Error on build

### ISSUE THREAD

**Issue by BrownieBits** [shounakmulay/Telephony#84]
I brought in the Telephony package, didnt use it anywhere, tried to build the project. I get the error below. Is this due to something in the latest version of flutter?

Note: Recompile with -Xlint:deprecation for details.
e: /Users/{user}/flutter/.pub-cache/hosted/pub.dartlang.org/telephony-0.1.3/android/src/main/kotlin/com/shounakmulay/telephony/sms/SmsMethodCallHandler.kt: (52, 62): Expecting a parameter declaration

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':telephony:compileDebugKotlin'.
> Compilation error. See log for more details

[✓] Flutter (Channel stable, 2.0.6, on macOS 11.3.1 20E241 darwin-x64, locale en-US)
    • Flutter version 2.0.6 at /Users/ibrown/flutter
    • Framework revision 1d9032c7e1 (2 weeks ago), 2021-04-29 17:37:58 -0700
    • Engine revision 05e680e202
    • Dart version 2.12.3

[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.3)
    • Android SDK at /Users/ibrown/Library/Android/sdk
    • Platform android-30, build-tools 29.0.3
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.5, Build version 12E262
    • CocoaPods version 1.10.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)

[✓] VS Code (version 1.56.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.22.0

[✓] Connected device (2 available)
    • SM T307U (mobile) • R52N607571J • android-arm64  • Android 10 (API 29)
    • Chrome (web)      • chrome      • web-javascript • Google Chrome 90.0.4430.212

• No issues found!

**Comment by Mabsten** [comment]
This is an issue related to kotlin version: in the file [android/build.gradle](https://github.com/shounakmulay/Telephony/blob/develop/android/build.gradle) of the package is setted the Kotlin version `1.4.10`; starting from Kotlin 1.4 the trailin commas are supported (like currently in Dart); so in the file android/src/main/kotlin/com/shounakmulay/telephony/sms/SmsMethodCallHandler.kt trailing commas are used. But making a new Flutter project with the Flutter Sdk 2.0 (or 2.2 too) the Kotlin version setted in build.gradle is `1.3.50`, which doesn't support trailing comma, so the trailing comma in SmsMethodCallHandler.kt is interpreted as a missing parameter.

TL;DR: in android/build.gradle set the Kotlin version `1.4.10` (or another version equal to or greater than 1.4) like in the telephony package.

Since kotlin, unlike Dart, is not used to bracking changes, I believe that upgrading the Kotlin version should not create problems for the other packages included in your project.

**Comment by BrownieBits** [comment]
Awesome that was the solve... thank you for getting back to me.

**Comment by shounakmulay** [comment]
Removed trailing comma in [v0.1.4](https://pub.dev/packages/telephony/versions/0.1.4)
